wezterm-gui/src/termwindow/mod.rs

```diff
@@ -1018,44 +1018,124 @@ impl TermWindow {
+                            .join(" "),
+                    ),
+                )?;
                 Ok(true)
             }
             WindowEvent::DroppedFile(paths) => {
                 let pane = match self.get_active_pane_or_overlay() {
                     Some(pane) => pane,
                     None => return Ok(true),
                 };
-                let paths = paths
-                    .iter()
-                    .map(|path| {
-                        self.config
-                            .quote_dropped_files
-                            .escape(&path.to_string_lossy())
-                    })
-                    .collect::<Vec<_>>()
-                    .join(" ")
-                    + " ";
-                pane.send_paste(&paths)?;
+                let paths_string = self.user_dropped_event( 
+                    "user-dropped-paths",
+                    paths
+                        .iter()
+                        .map(|path| path.to_string_lossy().to_string())
+                        .collect::<Vec<_>>(),
+                );
+                pane.send_paste(
+                    paths_string.as_deref().unwrap_or(
+                        (paths
+                            .iter()
+                            .map(|path| {
+                                self.config
+                                    .quote_dropped_files
+                                    .escape(&path.to_string_lossy())
+                            })
+                            .collect::<Vec<_>>()
+                            .join(" ")
+                            + " ")
+                            .as_str(),
+                    ),
+                )?;
                 Ok(true)
             }
             WindowEvent::DraggedFile(_) => Ok(true),
         }
     }
+    fn user_dropped_event(
+        &mut self, 
+        event_name: &str, 
+        data: Vec<String>
+    ) -> Option<String> {
+        match config::run_immediate_with_lua_config(|lua| {
+            if let Some(lua) = lua {
+                let tabs = self.get_tab_information();
+                let panes = self.get_pane_information();
+                let active_tab = tabs.iter().find(|t| t.is_active).cloned();
+                let active_pane = panes.iter().find(|p| p.is_active).cloned();
+                let (modifiers, _) = self.current_modifier_and_leds;
+
+                let v = config::lua::emit_sync_callback(
+                    &lua,
+                    (
+                        event_name.to_string(),
+                        (
+                            active_tab.clone(),
+                            active_pane.clone(),
+                            modifiers.to_string(),
+                            data.clone(),
+                        ),
+                    ),
+                )?;
+                match &v {
+                    mlua::Value::Nil => Ok(None),
+                    mlua::Value::Table(t) => {
```

Comment:
I'm not sure about handling the table case here, as it makes things more complex overall.
I think allowing for `Nil` or `String` only is fine, and other types will be implicitly `tostring` converted to string via https://docs.rs/mlua/latest/mlua/enum.Value.html#method.to_string